src/ElevatorSubsystem/ElevatorCar.java

```diff
@@ -0,0 +1,107 @@
+package ElevatorSubsystem;
+
+import common.messages.ElevatorMessage;
+import common.messages.ElevatorStatusResponse;
+import common.messages.Message;
+import common.messages.MessageChannel;
+import common.messages.MessageType;
+
+/**
+ * Entity representing an elevator car, contains composite subcomponents for major functionality.
+ * 
+ * @author Ryan Fife
+ *
+ */
+public class ElevatorCar implements Runnable {
+	private int id;
+	private boolean inService;
+	private MessageChannel messageChannel;
+	private ElevatorMotor motor;
+	private ElevatorDoor door;
+	
+	private final static double DOOR_SPEED = 3000;
+	private final static double ELEVATOR_SPEED = 0.5;
+	
+	public ElevatorCar(int id, MessageChannel messageChannel) {
+		this.id = id;
+		this.messageChannel = messageChannel;
+		this.inService = true;
+		door = new ElevatorDoor(DOOR_SPEED);
+		motor = new ElevatorMotor(ELEVATOR_SPEED);
+	}
+	
+	public int getId() {
+		return this.id;
+	}
+	
+	public ElevatorMotor getMotor() {
+		return motor;
+	}
+	
+	public ElevatorDoor getDoor() {
+		return door;
+	}
+	
+	public boolean getInService() {
+		return inService;
+	}
+	
+	public void setInService(boolean service) {
+		inService = service;
+	}
+
+	public void run() {
+		while(true) {
+			Message message = messageChannel.getMessage();
```

Comment:
As discussed, the elevator should send a ready request to the scheduler. When the scheduler receives the ready request, the scheduler will pass the job.